### PR TITLE
Added install and enable tests - part 2

### PR DESCRIPTION
**Pull request by b4handjr**
Found a problem in circleci that caused some weird hangs during the CI runs. It appeared that sometimes the ```@pytest.mark.skipif``` decorator would cause a problem. It could be that the ```set -ex``` bash script setting was causing the pytest script to exit when it parsed the decorator, if the arguments within ```@pytest.mark.skipif``` were incorrect.

See: https://www.gnu.org/software/bash/manual/bashref.html#The-Set-Builtin

### CODE REVIEW

**Comment by chuckharmston** [comment]
@jrbenny35 would you mind elaborating on what the bug was and why this fixed it?

**Comment by b4handjr** [comment]
> would you mind adding info about what the bug was and why this fixed it?

Sure, I didn't track it down too much but I think it was circle ci caching. I don't think it was a problem with the script files. That's why there were so many commits. 

I guess saying there was a bug with the test setup would be incorrect, more so, a quirk of circleci and how the new tests were being set up.

File: circle.yml [hunk 1]
```diff
@@ -17,7 +18,7 @@ machine:
 dependencies:
   cache_directories:
     - "addon/node_modules"
-    - "/home/ubuntu/firefox-downloads"
+    - # "/home/ubuntu/firefox-downloads"
```
Comment (fzzzy) [inline circle.yml]: I'm not super familiar with yaml -- what does this do? Exclude it from the list? Can we just delete the line instead?

File: circle.yml [hunk 2]
```diff
@@ -34,8 +35,8 @@ test:
     - ./bin/circleci/flake8.sh
     - mkdir integration-test-results
     - ./bin/circleci/test-firefox-release.sh
-    - ./bin/circleci/test-firefox-dev.sh
-    - # ./bin/circleci/test-firefox-nightly.sh
+    - # ./bin/circleci/test-firefox-dev.sh
```
Comment (fzzzy) [inline circle.yml]: Same here, can we just delete this?

File: frontend/test/ui/conftest.py [hunk 3]
```diff
@@ -17,11 +21,35 @@ def firefox_options(firefox_options):
...
     firefox_options.set_preference('extensions.webapi.testing', True)
-    firefox_options.set_preference('example.com', 'local')
+    firefox_options.set_preference('testpilot.env', 'local')
+    firefox_options.add_argument('-foreground')
     firefox_options.log.level = 'trace'
     return firefox_options
 
 
+@pytest.yield_fixture(scope='session', autouse=True)
+def enable_dev_experiment():
```
Comment (fzzzy) [inline frontend/test/ui/conftest.py:31]: This isn't called anywhere. I can kind of guess that the decorator automatically applies it, but can you clarify?

scope='session' means that it gets run once per session, and autouse means it gets used?

Does the code before the yield run at the beginning of the session, and the code after the yield run at the end of the session?

File: frontend/test/ui/requirements/requirements.txt [hunk 4]
```diff
@@ -1,7 +1,10 @@
 PyPOM==1.2.0
-pytest==3.2.1
+pytest==3.2.0
```
Comment (fzzzy) [inline frontend/test/ui/requirements/requirements.txt]: Is this intentional? What's the reason for the downgrade?

File: frontend/test/ui/test_install.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import os
+
+import pytest
+
+from pages.desktop.home import Home
+from pages.desktop.detail import Detail
+
+
+@pytest.mark.nondestructive
+@pytest.mark.skipif(os.environ.get('SKIP_INSTALL_TEST') is not None,
```
Comment (fzzzy) [inline frontend/test/ui/test_install.py:10]: Is there a technical reason for this skip?

**Review by fzzzy — CHANGES_REQUESTED** [review]
I just have some questions about the diff to clarify my own understanding. Thanks.

File: circle.yml (hunk 1 above)
Comment (b4handjr) [reply circle.yml]: I can remove this back to how it was. Nightly had a bug last week that caused it to not read the capabilities given to it by geckodriver.

File: circle.yml (hunk 2 above)
Comment (b4handjr) [reply circle.yml]: So just leave release and nightly?

File: frontend/test/ui/conftest.py (hunk 3 above)
Comment (b4handjr) [reply frontend/test/ui/conftest.py:31]: You are 100% correct.

File: frontend/test/ui/requirements/requirements.txt (hunk 4 above)
Comment (b4handjr) [reply frontend/test/ui/requirements/requirements.txt]: This wasn't intentional.

File: frontend/test/ui/test_install.py (hunk 5 above)
Comment (b4handjr) [reply frontend/test/ui/test_install.py:10]: Yes so the preference ```xpinstall.signatures.required', False``` cannot be set in firefox release or beta, but only dev and nightly. Currently dev, is being shipped as beta it seems, so this skips the install tests on release but runs them on nightly.

File: circle.yml (hunk 1 above)
Comment (fzzzy) [reply circle.yml]: Please do.

File: circle.yml (hunk 2 above)
Comment (fzzzy) [reply circle.yml]: I think release and nightly are good enough.

File: frontend/test/ui/test_install.py (hunk 5 above)
Comment (fzzzy) [reply frontend/test/ui/test_install.py:10]: Ok, great info, thanks.

**Comment by b4handjr** [comment]
@fzzzy Am I missing something?

**Comment by fzzzy** [comment]
@jrbenny35 Please squash it. Also it looks like the tests failed on the last commit.

**Comment by fzzzy** [comment]
Oh, it has conflicts, so it needs rebasing.

**Comment by b4handjr** [comment]
Should be good now!

**Comment by fzzzy** [comment]
Although the tests pass on this, once it is merged to master, the script which deploys to dev breaks in a strange way. I'm going to revert this until we figure out what is going on there.

**Pull request merged**